app/services/adjusted_fees/create_service.rb

```diff
@@ -17,22 +17,23 @@ def call
       charge = fee.charge
       return result.validation_failure!(errors: {charge: ['invalid_charge_model']}) if disabled_charge_model?(charge)
 
+      unit_precise_amount_cents = (params[:unit_precise_amount].presence || 0).to_f * fee.amount.currency.subunit_to_unit
```

Comment:
I guess we can simplify this a bit.
`.to_f` converts both `nil` and `""` to "0.0" which seems what you're trying to achieve.

```suggestion
      unit_precise_amount_cents = params[:unit_precise_amount].to_f * fee.amount.currency.subunit_to_unit
```